device-data/linking-events.md

```diff
@@ -1 +1,239 @@
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-06-14T10:52:45",
+  "time": "2016-06-14T17:52:45.845Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId",
+  "_groupId": "f4c834c27a",
+  "guid": "b6a3e6b7-d44f-44bb-810e-189b8f2b03c2",
+  "id": "aiipljcqrquq96pugkbr2bfj4mb4r2mv",
+  "createdTime": "2016-06-14T18:03:50.664Z",
+  "_version": 0,
+  "_active": true,
+  "_schemaVersion": 1
+}]
+```
+
+### Linking events through the new platform ingestion API
+
+Unlike the legacy jellyfish ingestion service, the new platform ingestion API creates GUIDs for the `id` of each event. This means that linked events such as pairs of `wizard` and `bolus` events **cannot** be uploaded separately. Instead, only the "outer" event–that is, the event that embeds another event that it wishes to link to–is uploaded with the "inner" linked event embedded inside it. Upon ingestion of this compound event, the new platform ingestion service:
+
+- generates the GUID for the inner event and stores it on that event's `id` field
+- extracts and stores the inner event (augmented with its newly-generated `id`) as an independent event
+- updates the appropriate field on the outer event to contain just the newly-generated GUID for the inner event instead of the entire object
+
+To demonstrate with the same example data as above, the data for ingestion via the new platform ingestion API looks like:
+
+```json
+[{
+  "type": "wizard",
+  "bgInput": 32,
+  "bgTarget": {
+    "target": 85,
+    "high": 145
+  },
+  "bolus": {
+    "type": "bolus",
+    "subType": "normal",
+    "normal": 1,
+    "clockDriftOffset": 0,
+    "conversionOffset": 0,
+    "deviceId": "DevId0987654321",
+    "deviceTime": "2016-06-14T10:52:45",
+    "time": "2016-06-14T17:52:45.845Z",
+    "timezoneOffset": -420,
+    "uploadId": "SampleUploadId"
+  },
+  "carbInput": 109,
+  "insulinCarbRatio": 6,
+  "insulinOnBoard": 20.032,
+  "insulinSensitivity": 6,
+  "recommended": {
+    "carb": 18.25,
+    "correction": 5.25,
+    "net": 3.5
+  },
+  "units": "mg/dL",
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-06-14T10:52:45",
+  "time": "2016-06-14T17:52:45.845Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId"
+}]
+```
+
+And the resulting data is nearly exactly the same, the only difference being that the `id`s on each object are [RFC 4122](https://tools.ietf.org/html/rfc4122 'RFC 4122: A Universally Unique IDentifier (UUID) URN Namespace') version 4 UUIDs with all `-`s stripped:
+
+```json
+[{
+  "type": "bolus",
+  "subType": "normal",
+  "normal": 1,
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-06-14T10:52:45",
+  "time": "2016-06-14T17:52:45.845Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId",
+  "_groupId": "f4c834c27a",
+  "guid": "4c2f3acc-8d1d-4df0-bc88-2fad64df8151",
```

Comment:
Yeah, we added `guid` when we wiped the slate clean data-wise and released BtUTC. We knew we would probably want to move towards true UUIDs and wanted to make sure we *had* one even for older data, since `id` is not one. We may want to phase out `guid` eventually if we end up not needing it, but I don't think we should be too hasty to do that.

If anyone thinks this explanation belongs in the doc itself, LMK and I'll expand it.